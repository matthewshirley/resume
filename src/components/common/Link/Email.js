import React from 'react';
import PropTypes from 'prop-types';

import Text from '../Text';

const EmailLink = ({ emailAddress }) => {
  if (!emailAddress) return (null);

  return (
    <Text as="a" href={`mailto:${emailAddress}`} m={0}>{emailAddress}</Text>
  );
};

EmailLink.propTypes = {
  emailAddress: PropTypes.string,
};

EmailLink.defaultProps = {
  emailAddress: null,
};

export default EmailLink;
