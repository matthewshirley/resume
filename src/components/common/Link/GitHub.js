import React from 'react';
import PropTypes from 'prop-types';

import Text from '../Text';

const GitHubLink = ({ github }) => {
  if (!github) return (null);

  return (
    <Text as="a" href={`https://github.com/${github}`} m={0}>{`https://github.com/${github}`}</Text>
  );
};

GitHubLink.propTypes = {
  github: PropTypes.string,
};

GitHubLink.defaultProps = {
  github: null,
};

export default GitHubLink;
