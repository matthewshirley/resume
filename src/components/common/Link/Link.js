import React from 'react';
import PropTypes from 'prop-types';

const Link = ({ href, children }) => {
  if (href) {
    return (
      <a href={href} target="_blank" rel="noopener noreferrer">
        {children}
      </a>
    );
  }

  return (children);
};

Link.propTypes = {
  children: PropTypes.element.isRequired,
  website: PropTypes.string,
};

Link.defaultProps = {
  website: null,
};

export default Link;
