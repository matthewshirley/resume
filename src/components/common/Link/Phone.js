import React from 'react';
import PropTypes from 'prop-types';

import Text from '../Text';

const PhoneLink = ({ phoneNumber }) => {
  if (!phoneNumber) return (null);

  return (
    <Text as="a" href={`tel:${phoneNumber}`} m={0}>{phoneNumber}</Text>
  );
};

PhoneLink.propTypes = {
  phoneNumber: PropTypes.string,
};

PhoneLink.defaultProps = {
  phoneNumber: null,
};

export default PhoneLink;
