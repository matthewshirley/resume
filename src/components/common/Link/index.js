export { default } from './Link';
export { default as EmailLink } from './Email';
export { default as PhoneLink } from './Phone';
