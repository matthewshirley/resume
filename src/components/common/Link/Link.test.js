import React from 'react';
import renderer from 'react-test-renderer';

import Link from './Link';

describe('Phone Link', () => {
  it('renders a link when a href is passed', () => {
    const tree = renderer
      .create(<Link href="google.com">Test</Link>)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('does not link without a href', () => {
    const tree = renderer
      .create(<Link href={null}>Test</Link>)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
