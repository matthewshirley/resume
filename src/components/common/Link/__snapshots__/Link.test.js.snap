// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Phone Link does not link without a href 1`] = `"Test"`;

exports[`Phone Link renders a link when a href is passed 1`] = `
<a
  href="google.com"
  rel="noopener noreferrer"
  target="_blank"
>
  Test
</a>
`;
